src/ladds/io/hdf5/HDF5Writer.cpp

```diff
@@ -96,11 +105,10 @@ void HDF5Writer::writeParticles(size_t iteration, const AutoPas_t &autopas) {
          static_cast<HDF5Definitions::FloatType>(vel[1]),
          static_cast<HDF5Definitions::FloatType>(vel[2])});
     vecId.emplace_back(id);
-    // track the highest particle id that was written to the file
-    // All particles that have a higher id than the highest id from the last time something was written are new
-    // and their static properties need to be recorded.
-    maxParticleId = std::max(maxParticleId, id);
-    if (maxWrittenParticleID == 0 or id > maxWrittenParticleID) {
+
+    // only add properties not yet added
+    if (addedConstantPropertiesIds.find(id) == addedConstantPropertiesIds.end()) {
+      addedConstantPropertiesIds.insert(id);
```

Comment:
isn't this potentially expensive? We do this if for every particle (O(N)) and the find is in O(log(M)) where `M==addedConstantPropertiesIds.size()` which is `>=N`.
For now I would be ok with merging this but please add a comment here that this might need improvement.

Reply:
Yes, it is :( I can add a comment